### PR TITLE
Fix #7422: Update playlist icons & url bar interactions

### LINKED ISSUE #7422: Playlist Revamp: URL bar icons & menus
See Figma: `iOS Playlist`

Should replace the icons used for playlist, along with change the popover when adding an item and showing a menu when tapping an added item that allows for quickly changing where the video is saved, remove/undo and opening in playlist

### PR DESCRIPTION
**Pull request by kylehickinson**
## Summary of Changes

This pull request fixes #7422 

Also updates all mentions of "Folder" with "Playlist" in the current version of Playlist

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

- New Popover when adding initially:
  - Verify tapping on the popover itself opens playlist
  - Verify tapping Change dismisses the popup and shows the change playlist UI
  - Verify after 4s the popover automatically dismisses
- Tapping on the icon after adding:
  - Verify remove/undo
  - Verify change playlist shows the change playlist UI
    - Verify changing the folder updates the text shown in the menu
  - Verify opening in playlist resumes at the current time in the web page

## Screenshots:

| | | |
| --- | --- | --- |
| Simulator Screenshot - iPhone 14 Pro - 2023-05-09 at 16 25 33 | Simulator Screenshot - iPhone 14 Pro - 2023-05-09 at 16 26 39 | Simulator Screenshot - iPhone 14 Pro - 2023-05-09 at 16 26 46 |
| Simulator Screenshot - iPhone 14 Pro - 2023-05-09 at 16 26 52 | Simulator Screenshot - iPhone 14 Pro - 2023-05-09 at 16 26 35 | Simulator Screenshot - iPhone 14 Pro - 2023-05-09 at 16 27 00 |

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/Brave/Frontend/Browser/Toolbars/UrlBar/PlaylistURLBarButton.swift [hunk 1]
```diff
@@ -7,63 +7,131 @@ import Foundation
...
+    if #available(iOS 16.0, *) {
+      action = UIAction(title: title, image: image, attributes: attributes.union(.keepsMenuPresented), handler: { _ in
+        let menu = handler()
+        self.menu = menu
+      })
+    } else {
+      action = UIAction(title: title, image: image, attributes: attributes, handler: UIAction.deferredActionHandler { _ in
+        let menu = handler()
+        self.menu = menu
+        self.contextMenuInteraction?.perform(NSSelectorFromString("_presentMenuAtLocation:"), with: CGPoint.zero)
```
Comment (kylehickinson) [inline Sources/Brave/Frontend/Browser/Toolbars/UrlBar/PlaylistURLBarButton.swift]: Forgot to obfuscate this

File: Sources/Brave/Frontend/Browser/Tab.swift [hunk 2]
```diff
@@ -407,6 +407,7 @@ class Tab: NSObject {
   }
 
   deinit {
+    assert(Thread.isMainThread)
```
Comment (kylehickinson) [inline Sources/Brave/Frontend/Browser/Tab.swift]: To be removed

File: Sources/Brave/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift [hunk 3]
```diff
@@ -293,6 +293,10 @@ class TabLocationView: UIView {
         self?.updateColors(isPrivateBrowsing)
       })
     
+    playlistButton.menuActionHandler = { action in
```
Comment (kylehickinson) [inline Sources/Brave/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift]: May need [unowned self]

**Review by Brandon-T — APPROVED** [review]
Works very well! Tested on iPhone and iPad :)

**Pull request merged**